src/test/VesperAdapterV1.t.sol

```diff
@@ -209,8 +213,18 @@ contract VesperAdapterV1Test is DSTestPlus {
 
         hevm.warp(block.timestamp + 10000000000);
         hevm.roll(block.number + 10000000000);
+
+        (address[] memory tokensDAI, uint256[] memory amountsDAI) = IVesperRewards(0x51EEf73abf5d4AC5F41De131591ed82c27a7Be3D).claimable(address(alchemistETHAddress));
+
+        IUniswapV3Factory factory = IUniswapV3Factory(uniSwapFactory);
+
+        IUniswapV3Pool pool = IUniswapV3Pool(factory.getPool(vspRewardToken, address(weth), uint24(3000)));
+        (uint160 sqrtPriceX96,,,,,,) =  pool.slot0();
+        uint256 priceWETH = uint(sqrtPriceX96) * (uint(sqrtPriceX96)) * (1e18) >> (96 * 2);
```

Comment:
just curious, where did this conversion formula come from?

Reply:
Uniswap pools store price oracles in slot0 which returns a bunch of different data. It happens to return the price in square root form as a uint160 so this is a simplified version of the conversion on the uniswap website.